pennylane/devices/qubit/measure.py

```diff
@@ -86,17 +87,28 @@ def csr_dot_products(
         TensorLike: the result of the measurement
     """
     total_wires = len(state.shape) - is_state_batched
-    Hmat = measurementprocess.obs.sparse_matrix(wire_order=list(range(total_wires)))
 
-    if is_state_batched:
+    if is_pauli_sentence(measurementprocess.obs):
+        Hmat = measurementprocess.obs.sparse_matrix(wire_order=list(range(total_wires)))
```

Comment:
could pull it out if the if-elif-else block now, but not super important.

Reply:
No, I should actually remove that line. Thanks for pointing it out.